server/plugin/permalinks_test.go

```diff
@@ -354,13 +354,13 @@ func TestMakeReplacements(t *testing.T) {
 			},
 		},
 		{
-			name:   "bad commit hash",
-			input:  "start https://github.com/mattermost/mattermost-server/blob/badhash/app/authentication.go#L15-L22 lorem ipsum",
-			output: "start https://github.com/mattermost/mattermost-server/blob/badhash/app/authentication.go#L15-L22 lorem ipsum",
+			name:   "link with branch name",
+			input:  "start https://github.com/mattermost/mattermost-server/blob/master/app/authentication.go#L15-L22 lorem ipsum",
+			output: "start \n[mattermost/mattermost-server/app/authentication.go](https://github.com/mattermost/mattermost-server/blob/master/app/authentication.go#L15-L22)\n```go\ntype TokenLocation int\n\nconst (\n\tTokenLocationNotFound TokenLocation = iota\n\tTokenLocationHeader\n\tTokenLocationCookie\n\tTokenLocationQueryString\n)\n```\n lorem ipsum",
```

Comment:
Does this test require connecting to the internet? Ideally this isn't the case, though idk if we have any sort of mocking in place for the requests to fetch code from GitHub. The other tests have a valid URL that points to this code, though this `master` branch one is a broken link. How does this test end up gathering this `TokenLocation` code for the test here?

- Working - https://github.com/mattermost/mattermost-server/blob/cbb25838a61872b624ac512556d7bc932486a64c/app/authentication.go#L15-L22
- Broken (used in this test) - https://github.com/mattermost/mattermost-server/blob/master/app/authentication.go#L15-L22

Reply:
@mickmister This test does not require internet connectivity. we have to mock in the place of request. [mocking](https://github.com/mattermost/mattermost-plugin-github/blob/4f19c5cd2580342cdb965d8e1b0d7d8285d449db/server/plugin/permalinks_test.go#L435)

the link is used in the mock function to return the intended value so it works on a broken link too.
Can update the link here but it will lead to breaking other test cases.